### ISSUE TITLE
Aliases not created properly - Missing filter information - ES 2.2.0

### ISSUE THREAD

**Issue by akgoel-mo** [elastic/elasticsearch#17234]
**Elasticsearch version**: 2.2.0

**JVM version**: 1.8.0_74

**OS version**: 3.13.0-74-generic #118-Ubuntu SMP (Ubuntu Server 14.04 LTS)

**Elasticsearch Client**: [elasticsearch-py](https://github.com/elastic/elasticsearch-py)

**Description of the problem including expected versus actual behavior**:
I am trying to add aliases with filters to some indexes. Basically I have write and read aliases on an index, with filter and routing specified. Based on some criteria, I need to modify the aliases to point to a new index, at which time, I remove the write alias from the existing index, move it to the newly created index and at the same add the same read alias to the newly created index. I am using update_aliases api in the python client to perform all three operations atomically.

*Expected behaviour*: The write alias should be removed from the old index and start pointing to the new index, while containing the filter and routing specified.

*Observed behaviour*: The write alias is removed from the old index, added to the new index but doesnt contain the specified filter. Attaching code snippets in steps to reproduce.

**Steps to reproduce**:
* Apply alias actions:
```
actions = {}
actions['actions'] = [{'remove': {'index': u'goibibo_201612', 'alias': u'w_goibibo-serviceresponsesuccess_201612'}}, {'add': {'filter': {'term': {'primaryName': u'goibibo-serviceresponsesuccess'}}, 'index': u'goibibo-serviceresponsesuccess_201612', 'alias': u'w_goibibo-serviceresponsesuccess_201612'}}, {'add': {'filter': {'term': {'primaryName': u'goibibo-serviceresponsesuccess'}}, 'index': u'goibibo-serviceresponsesuccess_201612', 'alias': u'r_goibibo-serviceresponsesuccess_201612'}}]
es.indices.update_aliases(body=actions)
```
* Fetch alias (should contain the filter, but doesnt)
```
es.indices.get_alias(name='w_goibibo-serviceresponsesuccess_201612')

{u'goibibo-serviceresponsesuccess_201612': {u'aliases': {u'w_goibibo-serviceresponsesuccess_201612': {}}}}
```

**Provide logs (if relevant)**:
Didn't see any error logs in master node

**Comment by javanna** [comment]
This is a duplicate of #16547, already fixed in 2.2.1. Thanks for reporting and for the nice recreation though!

**Comment by akgoel-mo** [comment]
Thanks @javanna. Will update my elasticsearch version.
